src/service/iot_config.proto

```diff
@@ -0,0 +1,396 @@
+//    https://github.com/helium/helium-crypto-rs/blob/main/src/public_key.rs#L347-L354
+//
+// == DevAddrs ==
+//
+// - `devaddr_ranges` are inclusive on both sides, `start_addr` and `end_addr`.
+//
+// - `org_res_v1.devaddr_ranges` provides the ranges of DevAddrs available to
+//    any `route_v1` under an Org.
+//
+// - `route_v1.devaddr_ranges` provides the ranges of DevAddrs that should go to
+//    this specific Route. This ranges must always fall within the
+//    `devaddr_ranges` of the owning org.
+
+enum action_v1 {
+  create = 0;
+  update = 1;
+  delete = 2;
+}
+
+// Define an organisation
+message org_v1 {
+  uint64 oui = 1;
+
+  // Org admin key
+  bytes owner = 2;
+  // Key only used for DC payments
+  bytes payer = 3;
+  // List of keys allowed some specific actions, see services.
+  repeated bytes delegate_keys = 4;
+  uint64 nonce = 5;
+}
+
+// Device address range, ex: 16#00000001 to 16#0000000A
+message devaddr_range_v1 {
+  uint32 start_addr = 1;
+  uint32 end_addr = 2;
+}
+
+// Device App EUI and Dev EUI
+message eui_v1 {
+  uint64 app_eui = 1;
+  uint64 dev_eui = 2;
+}
+
+// Packet Router protocol options (empty for now)
+message protocol_packet_router_v1 {}
+
+message protocol_gwmp_mapping_v1 {
+  region region = 1;
+  uint32 port = 2;
+}
+
+// GWMP protocol options (region to port mapping, see
+// https://github.com/Lora-net/packet_forwarder/blob/master/PROTOCOL.TXT)
+message protocol_gwmp_v1 { repeated protocol_gwmp_mapping_v1 mapping = 1; }
+
+// HTTP Roaming protocol options
+message protocol_http_roaming_v1 {
+  enum flow_type_v1 {
+    sync = 0;
+    async = 1;
+  }
+  flow_type_v1 flow_type = 1;
+  // milliseconds
+  uint32 dedupe_timeout = 2;
+  // path component of URL of roaming partner
+  string path = 3;
+}
+
+// Server Route definition
+message server_v1 {
+  // LNS address, ex: endpoint.test.com
+  string host = 1;
+  // LNS port, ex: 8080
+  uint32 port = 2;
+  oneof protocol {
+    protocol_packet_router_v1 packet_router = 3;
+    protocol_gwmp_v1 gwmp = 4;
+    protocol_http_roaming_v1 http_roaming = 5;
+  }
+}
+
+// Route definition
+message route_v1 {
+  // UUID
+  string id = 1;
+  // LoraWan Network ID
+  uint64 net_id = 2;
+  repeated devaddr_range_v1 devaddr_ranges = 3;
+  repeated eui_v1 euis = 4;
+  // Organization Unique ID
+  uint64 oui = 5;
+  server_v1 server = 6;
+  // Number of packet copies bought by this route
+  uint32 max_copies = 7;
+  // Incremental int for updates
+  uint64 nonce = 8;
+}
+
+// ------------------------------------------------------------------
+// Service Message Definitions
+// ------------------------------------------------------------------
+
+message org_list_req_v1 {}
+
+message org_list_res_v1 { repeated org_v1 orgs = 1; }
+
+message org_get_req_v1 { uint64 oui = 1; }
+
+message org_create_helium_req_v1 {
+  bytes owner = 1;
+  bytes payer = 2;
+  // Number of device address needed
+  uint64 devaddrs = 3;
+  uint64 timestamp = 4;
```

Comment:
Please document in seconds or milliseconds since epoch

Reply:
It is up there https://github.com/helium/proto/pull/238/files#diff-8ae91d2600ab1106932cfbec793495e92a68ff739eb3ede4934d6b3f94da38a3R17 is it enough?